app/v1/services/RetrieveCharitableGivingReliefService.scala

```diff
@@ -42,22 +42,33 @@ class RetrieveCharitableGivingReliefService @Inject() (connector: RetrieveCharit
       correlationId: String): Future[ServiceOutcome[RetrieveCharitableGivingReliefResponse]] = {
 
     val result = for {
-      responseWrapper <- EitherT(connector.retrieve(request)).leftMap(mapDownstreamErrors(desErrorMap))
+      responseWrapper <- EitherT(connector.retrieve(request)).leftMap(mapDownstreamErrors(downstreamErrorMap))
     } yield responseWrapper
 
     result.value
   }
 
-  private def desErrorMap =
-    Map(
-      "INVALID_NINO"            -> NinoFormatError,
-      "INVALID_TYPE"            -> InternalError,
-      "INVALID_TAXYEAR"         -> TaxYearFormatError,
-      "INVALID_INCOME_SOURCE"   -> InternalError,
-      "NOT_FOUND_PERIOD"        -> NotFoundError,
-      "NOT_FOUND_INCOME_SOURCE" -> NotFoundError,
-      "SERVER_ERROR"            -> InternalError,
-      "SERVICE_UNAVAILABLE"     -> InternalError
+  private def downstreamErrorMap: Map[String, MtdError] = {
+    val errors = Map(
+      "INVALID_NINO"                 -> NinoFormatError,
+      "INVALID_TYPE"                 -> InternalError,
+      "INVALID_TAXYEAR"              -> TaxYearFormatError,
+      "INVALID_INCOME_SOURCE"        -> InternalError,
+      "NOT_FOUND_PERIOD"             -> NotFoundError,
+      "NOT_FOUND_INCOME_SOURCE"      -> NotFoundError,
+      "SERVER_ERROR"                 -> InternalError,
+      "SERVICE_UNAVAILABLE"          -> InternalError
     )
 
+    val extraTysErrors = Map(
+      "INVALID_TAX_YEAR"             -> TaxYearFormatError,
+      "INVALID_CORRELATION_ID"       -> InternalError,
+      "INVALID_INCOMESOURCE_ID"      -> InternalError,
+      "INVALID_INCOMESOURCE_TYPE"    -> InternalError,
+      "SUBMISSION_PERIOD_NOT_FOUND"  -> NotFoundError,
+      "INCOME_DATA_SOURCE_NOT_FOUND" -> NotFoundError,
+      "TAX_YEAR_NOT_SUPPORTED"       -> RuleTaxYearNotSupportedError,
```

Comment:
`RuleTaxYearNotSupportedError` should really also be added to the Controller spec, in the "service errors" test